src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx

```diff
@@ -55,21 +67,52 @@ const newQuery = {
   conditions: '',
 };
 
+function mapErrors(
+  data: ValidationError,
+  update: FlatValidationError
+): FlatValidationError {
+  Object.keys(data).forEach((key: string) => {
+    const value = data[key];
+    // Recurse into nested objects.
+    if (Array.isArray(value) && typeof value[0] === 'string') {
+      update[key] = value[0];
+    } else if (Array.isArray(value) && typeof value[0] === 'object') {
+      update[key] = (value as ValidationError[]).map(item => mapErrors(item, {}));
+    } else {
+      update[key] = mapErrors(value as ValidationError, {});
+    }
+  });
+
+  return update;
+}
+
 class AddDashboardWidgetModal extends React.Component<Props, State> {
   state: State = {
     title: '',
     displayType: 'line',
     interval: '5m',
     queries: [{...newQuery}],
+    errors: undefined,
+    loading: false,
   };
 
   handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    const {closeModal, onAddWidget} = this.props;
-    onAddWidget(this.state as Widget);
-    addSuccessMessage(t('Added widget.'));
-    closeModal();
+    const {api, closeModal, organization, onAddWidget} = this.props;
+    this.setState({loading: true});
+    try {
+      const widgetData: Widget = omit(this.state, ['errors', 'loading']);
```

Comment:
Is there a reason to prefer `omit` over `pick` here? The [docs](https://lodash.com/docs/4.17.15#omit) mention that `omit` is considerably slower than `pick`.

Reply:
No real reason. I can use pick instead.

Reply:
Whoops, forgot to push the change with pick(). I'll do that in my next set of changes.